lib/discover/local.go

```diff
@@ -132,7 +132,7 @@ func (c *localClient) announcementPkt(msg []byte) ([]byte, bool) {
 	pkt := Announce{
 		ID:         c.myID,
 		Addresses:  addrs,
-		InstanceID: rand.Int63(),
+		InstanceID: instanceID,
 	}
```

Comment:
Why this change?

Reply:
Previously the packet was allocated outside of the loop, so got the instance ID at startup as was intended. My first change moved the randomness inside the loop, which is wrong. This fixes that.